backend/hostgw/hostgw_windows.go

```diff
@@ -13,14 +11,243 @@
+		if err != nil {
+			return nil, errors.Annotatef(err, "failed to create HNSNetwork %s", networkName)
+		}
+
+		// Wait for the network to populate Management IP
+		log.Infof("Waiting to get ManagementIP from HNSNetwork %s", networkName)
+		waitErr = wait.Poll(500*time.Millisecond, 5*time.Second, func() (done bool, err error) {
+			newNetwork, lastErr = hcsshim.HNSNetworkRequest("GET", newNetwork.Id, "")
+			return newNetwork != nil && len(newNetwork.ManagementIP) == 0, nil
+		})
+		if waitErr == wait.ErrWaitTimeout {
+			return nil, errors.Annotatef(lastErr, "timeout, failed to get management IP from HNSNetwork %s", networkName)
+		}
+
+		// Wait for the interface with the management IP
+		log.Infof("Waiting to get net interface for HNSNetwork %s (%s)", networkName, newNetwork.ManagementIP)
+		waitErr = wait.Poll(500*time.Millisecond, 5*time.Second, func() (done bool, err error) {
+			_, lastErr = netshHelper.GetInterfaceByIP(newNetwork.ManagementIP)
+			return lastErr == nil, nil
+		})
+		if waitErr == wait.ErrWaitTimeout {
+			return nil, errors.Annotatef(lastErr, "timeout, failed to get net interface for HNSNetwork %s (%s)", networkName, newNetwork.ManagementIP)
+		}
+
+		log.Infof("Created HNSNetwork %s", networkName)
+		expectedNetwork = newNetwork
+	}
+
+	// 5. Ensure a 1.2 endpoint on this network in the host compartment
+	createNewBridgeEndpoint := true
+	bridgeEndpointName := networkName + "_ep"
+	existingBridgeEndpoint, err := hcsshim.GetHNSEndpointByName(bridgeEndpointName)
+	if err == nil && existingBridgeEndpoint.IPAddress.String() == expectedPodGatewayAddress.String() {
+		log.Infof("Found existing bridge HNSEndpoint %s", bridgeEndpointName)
+		createNewBridgeEndpoint = false
+	}
+
+	// 6. Create a bridge HNSEndpoint
+	expectedBridgeEndpoint := existingBridgeEndpoint
+	if createNewBridgeEndpoint {
+		if existingBridgeEndpoint != nil {
+			if _, err = existingBridgeEndpoint.Delete(); err != nil {
+				return nil, errors.Annotatef(err, "failed to delete existing bridge HNSEndpoint %s", bridgeEndpointName)
+			}
+			log.Infof("Deleted stale bridge HNSEndpoint %s", bridgeEndpointName)
+		}
+
+		expectedBridgeEndpoint = &hcsshim.HNSEndpoint{
+			Name:           bridgeEndpointName,
+			IPAddress:      expectedPodGatewayAddress.ToIP(),
+			VirtualNetwork: expectedNetwork.Id,
+		}
+
+		log.Infof("Attempting to create bridge HNSEndpoint %+v", expectedBridgeEndpoint)
+		if existingBridgeEndpoint, err = expectedBridgeEndpoint.Create(); err != nil {
+			return nil, errors.Annotatef(err, "failed to create bridge HNSEndpoint %s", bridgeEndpointName)
+		}
+
+		log.Infof("Created bridge HNSEndpoint %s", bridgeEndpointName)
+	}
+
+	// Wait for the bridgeEndpoint to attach to the host
+	log.Infof("Waiting to attach bridge endpoint %s to host", bridgeEndpointName)
+	waitErr = wait.Poll(500*time.Millisecond, 5*time.Second, func() (done bool, err error) {
+		lastErr = expectedBridgeEndpoint.HostAttach(1)
```

Comment:
Should be:
lastErr = existingBridgeEndpoint.HostAttach(1)

Reply:
@ksubrmnn , on 197 line, `expectedBridgeEndpoint` is valued by `existingBridgeEndpoint`.

Reply:
@ksubrmnn , `expectedBridgeEndpoint` is valued by `existingBridgeEndpoint` on top.

Reply:
This is not true. expectedBridgeEndpoint.Create() returns the result of the HNSEndpoint created by hcsshim which is then stored in existingBridgeEndpoint. However, expectedBridgeEndpoint remains the same object as it is initialized in line 141 (without an ID).
existingBridgeEndpoint is an HNSEndpoint with the ID of the endpoint that is created. You need that ID for HNS to do HostAttach. As this code it, it fails with an HNS error, but with the suggested fix, it attaches the endpoint properly.

Reply:
Got, I will replace `expectedBridgeEndpoint` after `expectedBridgeEndpoint.Create()` in line 213 to satisfy that.